web/js/map/layerbuilder.js

```diff
@@ -214,64 +268,46 @@ export function mapLayerBuilder(models, config, cache, ui, store) {
    * @param {object} options - Layer options
    * @returns {object} OpenLayers WMTS layer
    */
-  var createLayerWMTS = function(def, options, day, state) {
+  const createLayerWMTS = function(def, options, day, state) {
     const activeDateStr = state.compare.isCompareA ? 'selected' : 'selectedB';
-    var proj, source, matrixSet, matrixIds, urlParameters, date, extent, start;
-    proj = state.proj.selected;
-    source = config.sources[def.source];
-    extent = proj.maxExtent;
-    start = [proj.maxExtent[0], proj.maxExtent[3]];
+    const proj = state.proj.selected;
+    const source = config.sources[def.source];
     if (!source) {
       throw new Error(def.id + ': Invalid source: ' + def.source);
     }
-    matrixSet = source.matrixSets[def.matrixSet];
+    const matrixSet = source.matrixSets[def.matrixSet];
     if (!matrixSet) {
       throw new Error(def.id + ': Undefined matrix set: ' + def.matrixSet);
     }
-    if (typeof def.matrixIds === 'undefined') {
-      matrixIds = [];
-      lodashEach(matrixSet.resolutions, function(resolution, index) {
-        matrixIds.push(index);
-      });
-    } else {
-      matrixIds = def.matrixIds;
-    }
-
-    if (day) {
-      if (day === 1) {
-        extent = [-250, -90, -180, 90];
-        start = [-540, 90];
-      } else {
-        extent = [180, -90, 250, 90];
-        start = [180, 90];
-      }
-    }
-
-    date = options.date || state.date[activeDateStr];
+    let date = options.date || state.date[activeDateStr];
     if (def.period === 'subdaily') {
       date = self.closestDate(def, options);
       date = new Date(date.getTime());
     }
     if (day && def.period !== 'subdaily') {
       date = util.dateAdd(date, 'day', day);
     }
-
-    urlParameters =
-      '?TIME=' + util.toISOStringSeconds(util.roundTimeOneMinute(date));
-    var sourceOptions = {
+    const { tileMatrices, resolutions, tileSize } = matrixSet;
+    const { origin, extent } = calcExtentsFromLimits(matrixSet, def.matrixSetLimits, day, proj);
+    const sizes = tileMatrices.map(({ matrixWidth, matrixHeight }) => [matrixWidth, -matrixHeight]);
+    const tileGridOptions = {
+      origin,
+      extent,
+      sizes,
+      resolutions,
+      matrixIds: def.matrixIds || resolutions.map((set, index) => index),
+      tileSize: tileSize[0]
+    };
+    const urlParameters = '?TIME=' + util.toISOStringSeconds(util.roundTimeOneMinute(date));
```

Comment:
Getting an error with this at the 10 minute interval: 'Uncaught RangeError: Invalid time value at Date.toISOString' `?v=-300.59814893574116,-248.4198550250752,129.57163784572364,92.67152243598143&t=2019-09-03-T11%3A34%3A48Z&ics=true&ici=5&icd=10&l=GOES-East_ABI_Band7_Thermal,GOES-East_ABI_Band13_Clean_Infrared,GOES-East_ABI_Band2_Red_Visible,GOES-East_ABI_Air_Mass(hidden),Reference_Labels(hidden),Reference_Features(hidden),Coastlines,VIIRS_SNPP_CorrectedReflectance_TrueColor(hidden),MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor`